rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,210 @@
+import logging
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovementCategory(Enum):
+    """Supported Asset Movement Types so far only deposit and withdrawals"""
+    DEPOSIT = auto()
+    WITHDRAWAL = auto()
+
+
+class AssetMovement(HistoryBaseEntry):
+    """Asset movement event representing deposits and withdrawals on exchanges."""
+
+    def __init__(
+            self,
+            timestamp: TimestampMS,
+            location: Location,
+            category: AssetMovementCategory,
+            asset: Asset,
+            balance: Balance,
+            identifier: int | None = None,
+            event_identifier: str | None = None,
+            unique_id: str | None = None,
+            extra_data: dict[str, Any] | None = None,
+            is_fee: bool = False,
+    ) -> None:
+        """
+        `category`: Type of asset movement
+        `unique_id`: Unique identifier for this asset movement.
+            Either the exchange transaction id or the associated onchain transaction hash.
+            Used in conjunction with location to generate the event identifier.
+        `is_fee`: Controls whether this represents a fee event.
+        """
+        location_name = get_formatted_location_name(location)
+        asset_symbol = asset.resolve_to_asset_with_symbol().symbol
+        if is_fee:
+            sequence_index = 1
+            event_type = HistoryEventType.DEPOSIT if category == AssetMovementCategory.DEPOSIT else HistoryEventType.WITHDRAWAL  # noqa: E501
+            event_subtype = HistoryEventSubType.FEE
+            notes = f'Pay {balance.amount} {asset_symbol} as {location_name} {str(category).lower()} fee'  # noqa: E501
+        else:
```

Comment:
With what I suggested just set event_type to the given event type and then depending on its value and the `is_fee` value set subtype, and notes.